cmd/snap-preseed/main.go

```diff
@@ -114,37 +109,37 @@ func run(parser *flags.Parser, args []string) (err error) {
 	}
 
 	if opts.Reset {
-		return resetPreseededChroot(chrootDir)
+		return preseed.ResetPreseededChroot(chrootDir)
 	}
 
 	var cleanup func()
 	if probeCore20ImageDir(chrootDir) {
-		var popts *PreseedOpts
-		popts, cleanup, err = prepareCore20Chroot(chrootDir)
+		var popts *preseed.PreseedOpts
+		popts, cleanup, err = preseed.PrepareCore20Chroot(chrootDir)
```

Comment:
it's a bit unclear to me why in the exported helpers we would keep Prepare* and Run* separated instead of having preseed.Classic(chrootDir) and preseed.Core20(chrootDir) doing each the whole thing for the respective case?

Reply:
No strong reason, I'll change that, thanks for the suggestion.

Reply:
Done (also un-exported some funcs/types).